packages/SequenceTranslator/src/model/parsing-validation/sequence-validator.ts

```diff
@@ -0,0 +1,52 @@
+import {NUCLEOTIDES} from '../const';
+import {MonomerLibWrapper} from '../monomer-lib/lib-wrapper';
+import {sortByReverseLength} from '../helpers';
+import {SYNTHESIZERS as FORMAT} from '../const';
+
+export class SequenceValidator {
+  constructor(private sequence: string) {
```

Comment:
I'm not sure this is the best pattern. Why is it a class and not just a couple of methods? Or something like this, if you really need it:

const validator = new SequenceValidator(format);
validator.validate(sequence
);